Document/0x05b-Basic-Security_Testing.md

```diff
@@ -74,58 +74,129 @@ As mentioned before, starting with Android 7, the Android OS will no longer trus
-The network security configuration uses an XML file where the app specifies which CA certificates will be trusted.
+From Android 7 onwards, the network security configuration allows apps to customize their network security settings, by defining which CA certificates the app will be trusting.
+
+In order to implement the network security configuration for an app, you would need to create a new xml resource file with the name `network_security_config.xml`. This is explained in detail in one of the [Google Android Codelabs](https://codelabs.developers.google.com/codelabs/android-network-security-config/#3 "Basic Network Security Configuration").
+
+After the creation, the apps must also include an entry in the manifest file to point to the new network security configuration file.
 
-res/xml/network_security_config.xml:
-```xml
-<?xml version="1.0" encoding="utf-8"?>
-<network-security-config>
-    <base-config>
-        <trust-anchors>
-            <certificates src="@raw/extracas"/>
-            <certificates src="system"/>
-        </trust-anchors>
-    </base-config>
-</network-security-config>
-```
-The apps must include an entry in the manifest file to point to the network security configuration file.
 ```xml
 <?xml version="1.0" encoding="utf-8"?>
 <manifest ... >
     <application android:networkSecurityConfig="@xml/network_security_config"
                     ... >
         ...
-    </application>decomp
+    </application>
 </manifest>
 ```
 
-In order to intercept the traffic of an application running on Android 7.0 and higher, you must follow the steps below:
+The network security configuration uses an XML file where the app specifies which CA certificates will be trusted. There are various ways to bypass the Network Security Configuration, which will be described below. Please also see the [Security Analyst’s Guide to Network Security Configuration in Android P](https://www.nowsecure.com/blog/2018/08/15/a-security-analysts-guide-to-network-security-configuration-in-android-p/ "Security Analyst’s Guide to Network Security Configuration in Android P") for further information.
 
-- Decompile the app using decompilation tools.[Manual static Analysis] provides details about decompiling (https://github.com/OWASP/owasp-mstg/blob/master/Document/0x05b-Basic-Security_Testing.md#manual-static-analysis)
-- Make the application trust the proxy's certificate by creating a network security configuration with the giving certificate as explained above
-- Repackage the app. The [Android developer documentation](https://developer.android.com/studio/publish/app-signing#signing-manually) explains how it's done.
+##### Adding the User Certificates to the Network Security Configuration
+
+There are different configurations available for the Network Security Configuration to [add non-system Certificate Authorities](https://developer.android.com/training/articles/security-config#CustomTrust "Custom Trust") via the src attribute:
```

Comment:
Can you add the tools and the links as references at the end?

Reply:
Added tools and references at the end